ct_mongodb

hDw1z opened the pull request:
Changes to be committed:
	new file:   ct_mongodb/LICENSE
	new file:   ct_mongodb/defaults/main.yml
	new file:   ct_mongodb/meta/main.yml
	new file:   ct_mongodb/tasks/main.yml

Commit:
ct_mongodb
Changes to be committed:
	new file:   ct_mongodb/LICENSE
	new file:   ct_mongodb/defaults/main.yml
	new file:   ct_mongodb/meta/main.yml
	new file:   ct_mongodb/tasks/main.yml

Commit:
ct_mongodb
Changes to be committed:
	modified:   README.md
	new file:   roles/ct_mongodb/README.md
	modified:   roles/ct_mongodb/defaults/main.yml
	modified:   roles/ct_mongodb/tasks/main.yml

Commit:
ct_mongodb
Changes to be committed:
	modified:   roles/ct_mongodb/tasks/main.yml

freehck commented on roles/ct_mongodb/README.md:
```diff
@@ -0,0 +1,29 @@
+
+# almaops.ct_mongodb
```
ты добавляешь роль в коллекцию almaops.common

Тут должно быть almaops.common.ct_mongodb

freehck commented on roles/ct_mongodb/README.md:
```diff
@@ -0,0 +1,29 @@
+  vars_files:
+    - "{{ playbook_dir }}/.../.yml" 
+  roles:
+    - role: "ct_mongodb"
```
аналогично: almaops.common.ct_mongodb

freehck commented:
@ussrlongbow 
можно было бы ещё добавить параметризацию `_evn`, но это в целом не обязательно
считаю, что можно мерджить

hDw1z commented:
Предлагаю пока не утверждать PR. Конфигурация ansible-роли ct_mongodb меняется. Пробую состыковать с ролью ct_rocketchat без дополнительной docker-сети.

ussrlongbow commented on roles/ct_mongodb/defaults/main.yml:
```diff
@@ -0,0 +1,34 @@
+  MONGODB_ENABLE_JOURNAL: "true"
+  # The value of "ALLOW_EMPTY_PASSWORD" should be exactly "yes". Otherwise the deployment will fail.
+  ALLOW_EMPTY_PASSWORD: "yes"
+# ct_mongodb_ct_bind_addr: "127.0.0.1"
```
Переменные вида `ct_mongodb_ct_*` определяют конфигурацию внутри контейнера.
Переменная `ct_mongodb_ct_bind_addr` не имеет смысла, потому что в контейнере приложение должно биндиться на 0.0.0.0

Эту переменную надо убрать

ussrlongbow commented on roles/ct_mongodb/meta/main.yml:
```diff
@@ -0,0 +1,18 @@
+# ct_mongodb
+
+
+
```
надо убрать лишние строки

ussrlongbow commented on roles/ct_mongodb/defaults/main.yml:
```diff
@@ -0,0 +1,34 @@
+
```
Все YAML файлы должны быть обособлены началом документа (`---`) и концом документа (`...`)
пример - https://github.com/almaops/ansible-collection-common/blob/master/roles/ct_minio/meta/main.yml#L1

ussrlongbow commented on roles/ct_mongodb/defaults/main.yml:
```diff
@@ -0,0 +1,34 @@
+ct_mongodb_task_prefix: "=== CT_MONGODB ==="
+
+# Path settings
+ct_mongodb_out_path_config: "/srv/{{ ct_mongodb_ct_name }}"
```
Что такое `_out_`?

Все переменные, которые содержат в себе пути на файловой системе сервера, где запускается контейнер, должны именоваться по маске `ct_mongodb_path_*`

Commit:
ct_mongodb
Changes to be committed:
	modified:   roles/ct_mongodb/README.md
	modified:   roles/ct_mongodb/defaults/main.yml
	modified:   roles/ct_mongodb/tasks/main.yml

hDw1z commented:
Замечания устранены. Заменил стандартный образ mongo на bitnami/mongodb, чтобы сделать его более универсальным, в частности, применимым к развертыванию Rocket.Chat'а.

ussrlongbow commented on roles/ct_mongodb/defaults/main.yml:
```diff
@@ -0,0 +1,36 @@
+ct_mongodb_ct_name: "mongodb"
+ct_mongodb_ct_volumes: "{{ ct_mongodb_out_path_config }}:/bitnami/mongodb"
+ct_mongodb_ct_env:
+  MONGODB_INITDB_ROOT_USERNAME: "root"
```
Переменные окружения надо вынести в файл
`templates/mongodb.env.j2`

ussrlongbow commented on roles/ct_mongodb/tasks/main.yml:
```diff
@@ -0,0 +1,32 @@
+  file:
+    path: "{{ ct_mongodb_out_path_config }}"
+    state: directory
+    owner: 1001
```
`owner` надо определить в переменной

ussrlongbow commented on roles/ct_mongodb/tasks/main.yml:
```diff
@@ -0,0 +1,32 @@
+
+
+- name: "{{ ct_mongodb_task_prefix }}. Creating the '{{ ct_mongodb_out_path_config }}' directory"
+  file:
```
Надо использовать FQCN - `ansible.builtin.file`

Commit:
ct_mongodb
Changes to be committed:
	modified:   roles/ct_mongodb/defaults/main.yml
	modified:   roles/ct_mongodb/meta/main.yml
	modified:   roles/ct_mongodb/tasks/main.yml
	new file:   roles/ct_mongodb/templates/mongodb.env.j2

hDw1z commented:
Замечания устранены.
Итоговый коммит
https://github.com/almaops/ansible-collection-common/pull/1/commits/96c531cefa2dbe5dc7f2c86b41feec0075ad46a7

ussrlongbow reviewed (approved):
looks nice, approved
